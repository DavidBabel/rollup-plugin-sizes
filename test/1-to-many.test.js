"use strict";

var rollup  = require("rollup").rollup,
    
    plugin = require("../index.js");

describe("rollup-plugin-sizes", () => {
    describe("single-entry, multiple-output bundle", () => {
        var { log, out } = require("./log.js")();

        it("should show basic output", () =>
            rollup({
                input : "./test/specimens/1-to-many/a.js",

                plugins : [
                    plugin({
                        log
                    })
                ]
            })
            .then((bundle) => Promise.all([
                bundle.generate({ format : "es" }),
                bundle.generate({ format : "cjs" })
            ]))
            .then(() => expect(out()).toMatchSnapshot())
        );

        it("should show detailed output ", () =>
            rollup({
                input : "./test/specimens/1-to-many/a.js",

                plugins : [
                    plugin({
                        log,

                        details : true
                    })
                ]
            })
            .then((bundle) => Promise.all([
                bundle.generate({ format : "es" }),
                bundle.generate({ format : "cjs" })
            ]))
            .then(() => expect(out()).toMatchSnapshot())
        );
    });
});
