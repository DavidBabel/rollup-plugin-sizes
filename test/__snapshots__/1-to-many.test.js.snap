// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rollup-plugin-sizes single-entry, multiple-output bundle should show basic output 1`] = `
"Details for \\"bundle:\\"
	bundle - 65 B (100.00%)
"
`;

exports[`rollup-plugin-sizes single-entry, multiple-output bundle should show detailed output  1`] = `
"Details for \\"bundle:\\"
	bundle - 65 B (100.00%)
		a.js - 44 B (67.69%)
		b.js - 21 B (32.31%)
"
`;
